src/lib/storage/vector_compression/simd_bp128/simd_bp128_iterator.hpp

```diff
@@ -18,77 +19,46 @@ class SimdBp128Iterator : public BaseCompressedVectorIterator<SimdBp128Iterator>
   using Packing = SimdBp128Packing;
 
  public:
-  SimdBp128Iterator(const pmr_vector<uint128_t>* data, const size_t size, const size_t absolute_index = 0u);
+  explicit SimdBp128Iterator(std::unique_ptr<SimdBp128Decompressor>&& decompressor, const size_t absolute_index = 0u)
```

Comment:
Add a comment on why this iterator is not really implemented, and is just a proxy for the decompressor?

Reply:
```
// The SimdBp128Iterators are very simple iterators that use the SimdBp128Decompressor to access single values via
  // dereference() as the compressor already has to the logic to cache/extract blocks/meta blocks.
```

Helpful?

Reply:
With that argumentation, we wouldn't need both iterators and accessors at all

Reply:
Well, they both serve a function. Not sure I understand what you are hinting at.

Reply:
Probably better to discuss this in person. It's more of an open question - I can't come up with a good comment myself.

Reply:
It's actually a difficult topic as some optimizations might make sense now, but won't once we fix the iterators. Let's discuss this in person tomorrow.